sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -283,6 +288,86 @@ export class ManagementClient extends LinkEntity {
       : undefined;
   }
 
+  // TO DO - Any better name??
+  private async _acquireLockHelper(
+    message_id: string | number | Buffer | undefined,
+    timeoutInMs: number
+  ): Promise<number> {
+    const initOperationStartTime = Date.now();
+    if (!this._isMgmtRequestResponseLinkOpen()) {
+      log.mgmt(
+        "[%s] Acquiring lock to get the management req res link.",
+        this._context.namespace.connectionId
+      );
+
+      const actionAfterTimeout = () => {
+        const desc: string = `The request with message_id "${message_id}" timed out. Please try again later.`;
+        const e: Error = {
+          name: "OperationTimeoutError",
+          message: desc
+        };
+
+        throw e;
+      };
+
+      const waitTimer = setTimeout(actionAfterTimeout, timeoutInMs);
+
+      try {
+        await defaultLock.acquire(this.managementLock, () => {
+          return this._init();
+        });
+      } catch (err) {
+        throw err;
+      } finally {
+        clearTimeout(waitTimer);
+      }
+    }
+    // Returns the time taken by the init operation
+    return Date.now() - initOperationStartTime;
+  }
+
+  // TO DO - Any better name??
+  private async _managementLinkSendRequestHelper(
+    request: AmqpMessage,
+    sendRequestOptions: SendRequestOptions
+  ): Promise<AmqpMessage> {
+    try {
+      return await this._mgmtReqResLink!.sendRequest(request, {
+        abortSignal: sendRequestOptions.abortSignal,
+        requestName: sendRequestOptions.requestName,
+        timeoutInMs: sendRequestOptions.timeoutInMs
+      });
+    } catch (err) {
+      err = translate(err);
+      const address =
+        this._mgmtReqResLink || this._mgmtReqResLink!.sender.address
+          ? "address"
+          : this._mgmtReqResLink!.sender.address;
+      log.warning(
+        "[%s] An error occurred during send on management request-response link with address " +
+          "'%s': %O",
+        this._context.namespace.connectionId,
+        address,
+        err
+      );
+      throw err;
+    }
+  }
+
+  private async _acquireLockAndSendRequest(
+    request: AmqpMessage,
+    retryTimeoutInMs: number,
+    sendRequestOptions: SendRequestOptions
+  ): Promise<AmqpMessage> {
+    const timeTakenByInit = await this._acquireLockHelper(request.message_id, retryTimeoutInMs);
+    const remainingOperationTimeoutInMs = retryTimeoutInMs - timeTakenByInit;
+    return await this._managementLinkSendRequestHelper(request, {
```

Comment:
It's generally better for performance/debuging purposes to not call `await` on a function if you're just going to return the value.
```suggestion
    return this._managementLinkSendRequestHelper(request, {
```
Better for debugging because now you don't have to worry about stepping into the tslib async/await pollyfill, and better for performance because otherwise V8 is going to wrap the value you awaited in a Promise since you're returning from an async function anyway. V8 doesn't wrap the value if it is already a promise.

It makes sense to await it in `_managementLinkSendRequestHelper` since you want to catch and handle the error that might result from the awaited call inside your function, but that's not happening in this one.

Reply:
Thanks for the info.. Makes a lot of sense.